Osgi support

atooni opened the pull request:
Hello, 

for now I have added the definitions for akka-actor-typed and akka-persistence-typed as we want to try that out in our container.

I have also fixed a broken import for akka-streams in 2.5.16

See #25327 and #25579 

There are no automated tests to verify that the bundles can be loaded by now, but I might be able to have a look at that another day by extracting some test cases from our framework.

ktoso committed:
+str #24812 fix signature of monitor()

* make monitor be a keep.both by default

patriknw committed:
Passivate for Typed Sharding, #24478

* also moved singleton doc sample to own file
* and mv /doc/ to /docs/ as in other places

richardimaoka committed:
Add more description about ClusterEvent.InitialStateAsSnapshot (Closes #25172)

Commit:
add zipLatest and zipLatestWith operators and graphstages

patriknw committed:
Simplify signature of ClusterSharding.spawn, #25480

patriknw committed:
Update version for release 2.5.16

patriknw committed:
Update version for release 2.5.16

patriknw committed:
Utility to check that same version of all modules is used

* Stolen from Cinnamon
* Can be used from outside of Akka, e.g. Akka HTTP or Lagom

johanandren committed:
Undeprecate the config, add a note in cluster singleton

patriknw committed:
Remove Scheduler param in EntityRef.ask, #25481

raboof committed:
BehaviorTestKit: get child inbox by anonymous ActorRef

patriknw committed:
ActorTestKit overhaul, #24598

* composition is the basic building block for ActorTestKit
* ActorTestKitWordSpec for integration with ScalaTest
  (automatic shutdown)

* Use ActorTestKitWordSpec in our own tests

* doc TestException

atooni committed:
Fixing imports for akka-streams

atooni committed:
Fixing imports for akka-streams

atooni committed:
adding OSGi definitions for akka-actor-typed

atooni committed:
adding OSGi definitions for akka-actor-typed

atooni committed:
Adding OSGi definitions for akka-persistence-typed

atooni committed:
Adding OSGi definitions for akka-persistence-typed

lightbend-cla-validator commented:
Hi @atooni,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

ktoso commented:
OK TO TEST

ktoso commented on akka-actor-tests/src/test/scala/akka/AkkaVersionSpec.scala:
```diff
@@ -59,7 +59,7 @@ class AkkaVersionSpec extends WordSpec with Matchers {
 
     "succeed if Akka version is SNAPSHOT" in {
-      AkkaVersion.require("AkkaVersionSpec", "2.5.6", "2.5-SNAPSHOT")
+      AkkaVersion.require("AkkaVersionSpec", "2.5.6", "2.5.16")
```
sure? the test is specifically about testing -SNAPSHOT

atooni replied:
Sorry, I had changed this to have a specific version in my OSGi Container to test with. I have now reverted that change.

ktoso replied:
No worries, thanks!

patriknw replied:
this is still remaining

atooni replied:
I have now changed that. However, I just double checked my git logs and it seems that I have never touched that file. I did at one point set 2.5.-SNAPSHOT to 2.5.16-OSGI. Nevermind...mysteries og Git :)

ktoso commented on project/AkkaBuild.scala:
```diff
@@ -28,7 +28,7 @@ object AkkaBuild {
     Protobuf.settings ++ Seq(
       parallelExecution in GlobalScope := System.getProperty("akka.parallelExecution", parallelExecutionByDefault.toString).toBoolean,
-      version in ThisBuild := "2.5-SNAPSHOT"
+      version in ThisBuild := "2.5.16"
```
Please revert this. Akka changes -SNAPSHOT things during releasing with scripts; this should not be changed like this in-place

ktoso commented on project/OSGi.scala:
```diff
@@ -102,6 +112,8 @@ object OSGi {
     imports = Seq(optionalResolution("org.fusesource.leveldbjni.*"), optionalResolution("org.iq80.leveldb.*")))
 
+  val persistenceTyped = exports(Seq("akka.persistence.typed.*"))
+
```
looks good, thanks

ktoso reviewed (changes requested):
Partially good, please revert the version number changes, the osgi changes are good I think. Thanks!

atooni committed:
Reverting version change

atooni committed:
Reverting version change

patriknw reviewed (commented):
Have you verified that the contents of the jar files are the same aside from the expected changes when doing `sbt publishLocal`. The reason I'm cautious is that there have been too many times the osgi build plugin (or whatever) breaks the artifacts.

atooni commented:
Hi, 

I did inspect the jar files and they look the same to me. Also I have tested my changes in our OSGi Container and the bundles seem to work as expected. When I find time, I might be able to create a further PR with a more sophisticated test. 

Best regards 
Andreas

patriknw committed:
hardening of CoordinatedShutdown init, #25592

patriknw committed:
Revert Support for rolling upgrade of ORSet[ActorRef], #25336

This reverts commit b72312d428223156c10130e19f921e8ae5f336b6.

patriknw committed:
Introduce RecipientRef interface for ActorRef and EntityRef, #24463

* to make it possible to use ActorContext.ask also with EntityRef
  destinations
* this abstraction will probably be useful for other things, where a
  full ActorRef is not required
* a RecipientRef only has to support tell (and ask), while a full ActorRef
  is watchable (destination has a lifecycle) and location transparent (serializable)

patriknw reviewed (commented):
Thanks for verifying, one thing remaining...

atooni committed:
Reverting version change

atooni committed:
Reverting version change

chbatey committed:
make scalatest suite type agnostic

patriknw commented on project/OSGi.scala:
```diff
@@ -102,6 +112,8 @@ object OSGi {
     Seq("akka.persistence.*"),
     imports = Seq(optionalResolution("org.fusesource.leveldbjni.*"), optionalResolution("org.iq80.leveldb.*")))
 
+  val persistenceTyped = exports(Seq("akka.persistence.typed.*"))
```
isn't this needed for the other -typed modules also?
* actorTestkitTyped
* clusterTyped
* clusterShardingTyped
* streamTyped

atooni replied:
Hi, 
I have just added what I actually required. In principle I would agree except for the testkit unless someone wanted to use the testkit inside an OSGi enabled container. 

With respect to the others it might be better to have some kind of validation to make sure that the jar file contains what you expect. I might be able to invest some time in a raw validity check that is better than nothing next week. I am on site all week this week. 

Rgds
Andreas

patriknw committed:
Some more BehaviorIntercept tests

* filter messages
* show how it can be used to implement PoisonPill

patriknw committed:
Merge pull request #25586 from akka/wip-24598-testkit-patriknw

ActorTestKit overhaul, #24598

jrudolph committed:
Docs: don't use $code$ for now in doc snippets

patriknw committed:
page about rolling updates, and change log

patriknw committed:
Merge pull request #25638 from jrudolph/fix-docs

Docs: don't use $code$ for now in doc snippets

patriknw committed:
Merge pull request #25633 from akka/wip-more-intercept-test-patriknw

Some more BehaviorIntercept tests

patriknw committed:
Merge pull request #25572 from akka/wip-25481-EntityRef-scheduler-patriknw

Remove Scheduler param in EntityRef.ask, #25481

patriknw committed:
Merge pull request #25563 from akka/wip-manifest-version-patriknw

Utility to check that same version of all modules is used

JustinPihony committed:
Add deprecation documentation to bin-compat docs (#25400)

* Add deprecation documentation to bin-compat docs

Closes #25399

* Be more flexible and change version type

patriknw committed:
Use supervision in all places of Source.fromIterator, #25574 (#25601)

* it was noticed in Source.fromIterator depending on where the
  iterator throwed exception
* fromIterator, as many other things, is implemented with
  statefulMapConcat
* supervision was only used for exceptions in onPush and not
  in onPull, added it there also

johanandren committed:
Clarify that there should normally be only one ActorSystem (#25641)

patriknw committed:
Improve AskTimeoutException message, #25644

chbatey committed:
wip

ktoso committed:
add TODOs to encourage contributing operator examples (#25478)

ktoso committed:
=doc #24283 include rationale for Sink Source not extending RS (#24286)

* Document reasons why Akka Streams are not IS-A with RS interfaces

* amended

johanandren committed:
Merge pull request #25565 from johanandren/wip-down-removal-margin-take-two-johanandren

Undeprecate the config, add a note in cluster singleton

ktoso committed:
add usage examples for conflate / conflate with seed (#25498)

* +doc conflate and conflateWithSeed docs examples

* fixes

* show throttle in the example so its more obvious that "slow upstream" etc

chbatey committed:
Reset timeout for fixed restarts

patriknw committed:
Merge pull request #25225 from akka/wip-24478-passivate-patriknw

WIP: Passivate for Typed Sharding, #24478

chbatey committed:
Remove old restarters

raboof commented:
PLS BUILD

chbatey committed:
Dropped messages when backing off

johanandren committed:
Remove the ActorContext parameter from the scaladsl command handler #25620

chbatey committed:
Private all the things

johanandren committed:
ActorContext removed from more signatures #25620

johanandren committed:
Command context no longer needed

johanandren committed:
Docs sample of accessing context from persistent behavior

johanandren committed:
Remove CommandHandler.byState #25655

chbatey committed:
Review feedback

chbatey committed:
Review: Don't double log exception

raboof committed:
Add troubleshooting hint to logging

johanandren committed:
Some usages of the removed signatures

Commit:
Protobuf serializers for ShardRegionStats #25348 #23250 #24465

johanandren committed:
Merge pull request #25654 from johanandren/wip-25620-remove-typed-persistence-ctx-param-johanandren

Remove the ActorContext parameter from the scaladsl command handler

patriknw committed:
Merge pull request #25539 from akka/wip-25480-sharding-start-patriknw

Simplify signature of ClusterSharding.spawn, #25480

patriknw committed:
Merge pull request #25610 from akka/wip-24463-EntityRef.ask-patriknw

Introduce RecipientRef interface for ActorRef and EntityRef, #24463

chbatey committed:
Warning for truncated DNS messages (#25666)

* Log warning if DNS response truncated
* Docs msg

johanandren committed:
Disallow nested mixed widen usages, #25604

johanandren committed:
Deduplicate is specific to interception so belongs where it is #25599 (#25665)

patriknw committed:
AccountExample in javadsl, #25485 (#25505)

johanandren committed:
Merge pull request #25653 from chbatey/supervision-interceptor

Supervision with a BehaviorInterceptor

patriknw committed:
Don't automatically down quarantined node, #25632

tz70s committed:
Add and reuse example for FileIO stream operator. (#25660)

chbatey committed:
Only treat nameservers as a StringList in config if getting as String… (#25636)

* Only treat nameservers as a StringList in config if getting as String fails

chbatey committed:
Provide minSequenceNr for snapshot deletion (#25590)

* Provide minSequenceNr for snapshot deletion

Journals can use this to make the bulk deletion more efficient

Use keepNrBatches to delete the last few snapshots in case previous
deletes failed.

patriknw committed:
Merge pull request #25679 from akka/wip-monitor-api-patriknw

+str #24812 fix signature of monitor()

cjtraverso committed:
Update Akka Cluster Bootstrap url

Currently the url is: https://developer.lightbend.com/docs/akka-management/current/bootstrap.html
It should be changed to: https://developer.lightbend.com/docs/akka-management/current/bootstrap/index.html

cjtraverso committed:
Update Akka Bootstrap Url

patriknw committed:
Revert "Avoid ask to sharding guardian if region already cached", #25683

* revert a7656ab8c4a93a3fe8af7a22690cb33f7a280af5
* test for reproducing the issue with computeIfAbsent
* use non-blocking get as optimization for avoiding ask

johanandren committed:
Merge pull request #25682 from cjtraverso/patch-1

Update Akka Bootstrap URL

patriknw committed:
Merge pull request #25645 from akka/wip-AskTimeoutException-patriknw

Improve AskTimeoutException message, #25644

patriknw committed:
Merge pull request #25678 from akka/wip-25632-down-Terminated-patriknw

Don't automatically down quarantined node, #25632

patriknw committed:
Merge pull request #25684 from akka/wip-25683-revert-computeIfAbsent-patriknw

Revert "Avoid ask to sharding guardian if region already cached", #25683

patriknw committed:
Merge pull request #25595 from akka/wip-25336-orset-actorref-patriknw

Revert Support for rolling upgrade of ORSet[ActorRef], #25336

patriknw committed:
thenReply Effect, #25482

* enforce ReplyEffect

chbatey committed:
Upgrade to scala 2.12.7

chbatey committed:
Mima latest version to 2.5.17

patriknw committed:
rename PersistentBehaviors.receive to PersistentBehavior.apply

raboof committed:
Support IPv6 in async-dns (#25687)

* Support IPv6 in async-dns

* Expect AAAA records for IPv6

johanandren committed:
Document MutableBehavior  (#25685)

* Document MutableBehavior #25631

patriknw committed:
PersistentEntity to glue together Sharding and PersistentBehavior better

* Makes the combination more visable
* You don't have to worry about the persistenceId, only EntityTypeKey and entityId
* The glue is stronger in the javadsl because of two reasons
  * Couldn't realisticly create a PersistentEntity class extending PersistenBehavior (which
    contains all the optional parameters and functions) since that would duplicate too much.
  * The ActorContext would be needed in the ShardedEntityContext parameter and because of the
    additional M type parameters the type inference breaks down when using the factory. Would
    require specifying the type of the ShardedEntityContex[M] parameter. That problem doesn't
    seem to exist in Java.

renamed:
s/ShardedEntityContext/EntityContext/
s/ShardedEntity/Entity/

johanandren committed:
Merge pull request #25693 from chbatey/scala2127

Upgrade to scala 2.12.7

nvollmar committed:
Removing deprecation mark for method without maximum restart attempts #25714

vhiairrassary committed:
Remove duplicated section in doc

raboof committed:
Specify the artery port '0' to avoid address already in use

patriknw committed:
Merge pull request #25694 from chbatey/mima2517

Mima latest version to 2.5.17

raboof committed:
Fix parsing truncated DNS messages (#25691)

* Fix parsing truncated DNS messages

* Additional validations

salehkhazaei committed:
Add examples for zipWithIndex #25468 (#25661)

* Add examples for zipWithIndex #25468

* better examples

johanandren committed:
Merge pull request #25723 from akka/arteryTests

Specify the artery port '0' to avoid address already in use

Commit:
change semantics of ZipLatest slightly to account for pending first tuple with subsequent completion

Commit:
add additional coverage for possible case

Commit:
support up to 22 inputs as per spec and fix race in test

Commit:
fix similar race in another test

johanandren committed:
Merge pull request #25715 from nvollmar/master

Removing deprecation mark for method without maximum restart attempts #25714

johanandren committed:
Merge pull request #25716 from vhiairrassary/patch-2

Remove duplicated section in doc

johanandren committed:
Merge pull request #25538 from jchapuis/zipLatest

zipLatest and zipLatestWith operators (refs #25392)

chbatey committed:
Treat MemberStatus.Removed as terminal state in ClusterReadView (#25499)

* Fixes #25489 where cluster event for a previous state can override
the call to cluster.close settings it to remove
* Fix case where Removed is used as a placeholder for unknown

patriknw committed:
Merge pull request #25593 from akka/wip-25592-terminated-patriknw

hardening of CoordinatedShutdown init, #25592

mattkohl committed:
Documentation fixes (#25725)

* tread -> thread

chbatey committed:
Upgrade to Aeron 1.11

raboof committed:
Don't warn about old DNS protocol use

We use the old DNS protocol ourselves, and the old implementation (which is
still the default) does not support the new protocol. For now the old protocol
is fine and works with both the default and the async impl.

patriknw committed:
Merge pull request #25743 from chbatey/aeron1-11

Upgrade to Aeron 1.11

andreaTP committed:
[=doc] Fix Akka Typed fiddle (#25754)

dwijnand committed:
Upgrade ssl-config from 0.2.4 to 0.3.4 (#25707)

chbatey committed:
Clarify supervision of singleton actors (#25681)

* Clarify supervision of singleton actors
* Document supervision for typed singletons

ollyw committed:
Fix doc links to cluster docker compose sample (#25756)

mehdimousavi1995 committed:
example for reduce operator #25468

dwijnand committed:
Upgrade to ssl-config 0.3.5, from 0.3.4

These are the additional binary API changes introduced:

    [error]  * class com.typesafe.sslconfig.ssl.FakeKeyStore is declared final in current version
    [error]    filter with: ProblemFilters.exclude[FinalClassProblem]("com.typesafe.sslconfig.ssl.FakeKeyStore")

    [error]  * method GeneratedKeyStore()java.lang.String in class com.typesafe.sslconfig.ssl.FakeKeyStore does not have a correspondent in current version
    [error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("com.typesafe.sslconfig.ssl.FakeKeyStore.GeneratedKeyStore")

    [error]  * method SignatureAlgorithmOID()sun.security.util.ObjectIdentifier in class com.typesafe.sslconfig.ssl.FakeKeyStore does not have a correspondent in current version
    [error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("com.typesafe.sslconfig.ssl.FakeKeyStore.SignatureAlgorithmOID")

    [error]  * method SignatureAlgorithmName()java.lang.String in class com.typesafe.sslconfig.ssl.FakeKeyStore does not have a correspondent in current version
    [error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("com.typesafe.sslconfig.ssl.FakeKeyStore.SignatureAlgorithmName")

patriknw committed:
Merge pull request #25696 from akka/wip-PersistentBehavior-rename-patriknw

rename PersistentBehaviors.receive to PersistentBehavior.apply

He-Pin committed:
update to sbt 1.2.4 (#25759)

patriknw committed:
Merge pull request #25764 from dwijnand/ssl-config

Upgrade to ssl-config 0.3.5, from 0.3.4

johanandren committed:
Cleanup typed test output (#25669)

* Cleanup ActorContextSpec output

* Default to not log dead letters and hide up to info level from tests

* Use log instead of println to benefit from config to silence

* The silence of the AdapterSpec

* Hushed the AskSpec

* Subdue the TimerSpec

* Muffled the FaultToleranceDocTest

* Remove the application.conf from the test in waiting for #25708

ignasi35 committed:
Bumps ssl-config 0.3.6

johanandren committed:
Max number of restarts for typed supervision backoff #25584 (#25772)

johanandren committed:
Docs for typed distributed data (#25747)

* Typed distributed data docs shaping up

Prsna23 committed:
Modified iff statements from JavaDSL/Flow.scala (#25785)

johanandren committed:
Java API checklist (#25762)

octonato committed:
fixed typo in DNS extension documentation (#25789)

patriknw committed:
Merge pull request #25625 from btomala/feature/25348_missing_serializers_for_shard_region

Protobuf serializers for ShardRegionStats

patriknw committed:
Merge pull request #25744 from akka/dontWarnAboutOldProtocolUse

Don't warn about old DNS protocol use

markehammons committed:
Don't return `Iterable(null)` when Probe.fishForMessage times out (#25629)

* Fix for issue #25616

I haven't had a chance to test this, but it looks to me that this is the solution to issue #25616. 

My stacktrace in this issue is this: 
```
java.lang.NullPointerException
	at eu.phy_awaas.irods.typed.lockingsystem.IRODSLockManagerSpec.$anonfun$new$20(IRODSLockManagerSpec.scala:91)
	at akka.actor.testkit.typed.internal.TestProbeImpl.loop$1(TestProbeImpl.scala:189)
	at akka.actor.testkit.typed.internal.TestProbeImpl.loop$1(TestProbeImpl.scala:201)
	at akka.actor.testkit.typed.internal.TestProbeImpl.fishForMessage_internal(TestProbeImpl.scala:215)
	at akka.actor.testkit.typed.internal.TestProbeImpl.fishForMessage_internal(TestProbeImpl.scala:45)
	at akka.actor.testkit.typed.scaladsl.TestProbe.fishForMessage(TestProbe.scala:175)
	at akka.actor.testkit.typed.scaladsl.TestProbe.fishForMessage$(TestProbe.scala:174)
	at akka.actor.testkit.typed.internal.TestProbeImpl.fishForMessage(TestProbeImpl.scala:45)
	at akka.actor.testkit.typed.scaladsl.TestProbe.fishForMessage(TestProbe.scala:181)
	at akka.actor.testkit.typed.scaladsl.TestProbe.fishForMessage$(TestProbe.scala:180)
	at akka.actor.testkit.typed.internal.TestProbeImpl.fishForMessage(TestProbeImpl.scala:45)
	at eu.phy_awaas.irods.typed.lockingsystem.IRODSLockManagerSpec.$anonfun$new$18(IRODSLockManagerSpec.scala:88)
	at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:1078)
	at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
	at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
	at org.scalatest.WordSpec.withFixture(WordSpec.scala:1881)
	at org.scalatest.WordSpecLike.invokeWithFixture$1(WordSpecLike.scala:1076)
	at org.scalatest.WordSpecLike.$anonfun$runTest$1(WordSpecLike.scala:1088)
	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:289)
	at org.scalatest.WordSpecLike.runTest(WordSpecLike.scala:1088)
	at org.scalatest.WordSpecLike.runTest$(WordSpecLike.scala:1070)
	at org.scalatest.WordSpec.runTest(WordSpec.scala:1881)
	at org.scalatest.WordSpecLike.$anonfun$runTests$1(WordSpecLike.scala:1147)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:396)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:373)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:410)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:373)
	at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:410)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:384)
	at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:379)
	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:461)
	at org.scalatest.WordSpecLike.runTests(WordSpecLike.scala:1147)
	at org.scalatest.WordSpecLike.runTests$(WordSpecLike.scala:1146)
	at org.scalatest.WordSpec.runTests(WordSpec.scala:1881)
	at org.scalatest.Suite.run(Suite.scala:1147)
	at org.scalatest.Suite.run$(Suite.scala:1129)
	at org.scalatest.WordSpec.org$scalatest$WordSpecLike$$super$run(WordSpec.scala:1881)
	at org.scalatest.WordSpecLike.$anonfun$run$1(WordSpecLike.scala:1192)
	at org.scalatest.SuperEngine.runImpl(Engine.scala:521)
	at org.scalatest.WordSpecLike.run(WordSpecLike.scala:1192)
	at org.scalatest.WordSpecLike.run$(WordSpecLike.scala:1190)
	at eu.phy_awaas.irods.typed.lockingsystem.IRODSLockManagerSpec.org$scalatest$BeforeAndAfterAll$$super$run(IRODSLockManagerSpec.scala:22)
	at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
	at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
	at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
	at eu.phy_awaas.irods.typed.lockingsystem.IRODSLockManagerSpec.run(IRODSLockManagerSpec.scala:22)
	at org.scalatest.tools.SuiteRunner.run(SuiteRunner.scala:45)
	at org.scalatest.tools.Runner$.$anonfun$doRunRunRunDaDoRunRun$13(Runner.scala:1340)
	at org.scalatest.tools.Runner$.$anonfun$doRunRunRunDaDoRunRun$13$adapted(Runner.scala:1334)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at org.scalatest.tools.Runner$.doRunRunRunDaDoRunRun(Runner.scala:1334)
	at org.scalatest.tools.Runner$.$anonfun$runOptionallyWithPassFailReporter$24(Runner.scala:1031)
	at org.scalatest.tools.Runner$.$anonfun$runOptionallyWithPassFailReporter$24$adapted(Runner.scala:1010)
	at org.scalatest.tools.Runner$.withClassLoaderAndDispatchReporter(Runner.scala:1500)
	at org.scalatest.tools.Runner$.runOptionallyWithPassFailReporter(Runner.scala:1010)
	at org.scalatest.tools.Runner$.run(Runner.scala:850)
	at org.scalatest.tools.Runner.run(Runner.scala)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.runScalaTest2(ScalaTestRunner.java:131)
	at org.jetbrains.plugins.scala.testingSupport.scalaTest.ScalaTestRunner.main(ScalaTestRunner.java:28)
```

With the calling code looking like this:

```scala
        val lockResults =
          probe.fishForMessage(Duration.create(10000, "ms"))(l => {
            r += 1
            if (r < 5) {
              l.lock.use(())//crash happens here
              FishingOutcomes.continue
            } else {
              l.lock.use(())
              FishingOutcomes.complete
            }
          })
```

It's pretty obvious receiveOne(timeout) returns a null when it receives nothing before the timeout, passes that along to my fisher function, which tries to perform work on the supposed message only to get a null instead.

* Added missing brace

* Manually formatted my code the way akka likes

I don't have the set up to compile this right now, so I just manually inserted the formatting that jenkins mentioned.

* More changes to fit the akka formatting standard

* Test for the new assertion error when fishing times out.

* Cleaned up a flaw I added from when I initially put this in async spec

skitt committed:
Avoid incorrect OSGi dep on sslconfig.akka (#25580)

Importing com.typesafe.sslconfig.* with the sslconfig version results
in OSGi bundle imports which also include com.typesafe.sslconfig.akka,
which can’t be satisfied with the given version range. Splitting up
the imported packages (com.typesafe.sslconfig,
com.typesafe.sslconfig.ssl.*, com.typesafe.sslconfig.util.*) results
in tighter-scoped imports which can be satisfied.

This fixes #25579.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

raboof committed:
Document 'withBackoff' srouce/flow/sink (#25770)

patriknw committed:
Use Props factory methods in getting started tutorial (#25713)

* Use Props factory methods in getting started tutorial

* deprecate Props.create without actorClass parameter, #25718

jayeve committed:
grammar nit (#25791)

patriknw committed:
make java sample code independent of Scala (#25800)

* need this for gathering and compiling java sources (in IDE),
  https://github.com/patriknw/akka-compile-ide-java

raboof committed:
Async DNS over TCP (#25690)

He-Pin committed:
=plugin migrate FastClasspathScanner to ClassGraph (#25797)

patriknw committed:
Rename MutableBehavior to AbstractBehavior, #25750

* Also cleanup javadsl Receive, which is only used from AbstractBehavior
* Clarify further in docs that the functional vs OO style is a matter of taste

patriknw committed:
Merge pull request #25577 from akka/getAnonymousChildInboxByActorRef

BehaviorTestKit: get child inbox by anonymous ActorRef

patriknw committed:
update to sbt 1.2.6

* due to https://github.com/sbt/zinc/pull/606

patriknw committed:
Merge pull request #25805 from akka/wip-sbt-1.2.6-patriknw

update to sbt 1.2.6

patriknw committed:
Merge pull request #25783 from ignasi35/bumps-ssl-config-036

Bumps ssl-config 0.3.6

patriknw committed:
PersistenceId type to differentiate between persistenceId and entityId, #25703 (#25704)

* PersistenceId type to differentiate between persistenceId and entityId, #25703

* both entityId (for sharding) and persistenceId as String types was easy
  mix-up
* utility method in EntityTypeKey to concatenaty the type and entityId to
  a unique persistenceId

* support custom separator to enable compatilbility with Lagom's javadsl

patriknw committed:
remove verbose from classgraph (#25806)

* this also fixes the performance regression of akka-docs/paradox

patriknw committed:
Merge pull request #25692 from akka/wip-25482-then-reply-patriknw

thenReply Effect, #25482

patriknw committed:
AccountExample in Scala in a few flavors, #25485

* AccountExample in Scala in a few flavors
* include Account examples in reference docs
* cleanup BlogPost example
* include reply doc snippets

patriknw committed:
null as empty state, in javadsl, #25768

patriknw committed:
Merge pull request #25697 from akka/wip-PersistentEntity-patriknw

PersistentEntity to glue together Sharding and PersistentBehavior better

patriknw committed:
Merge pull request #25804 from akka/wip-25750-AbstractBehavior-patriknw

Rename MutableBehavior to AbstractBehavior, #25750

Dollyg committed:
Create typed ActorMaterializer from ActorContext, ##25536

* Create typed ActorMaterializer from ActorContext to bind stream's lifecycle with an actor's lifecycle

* Add Java API for creating typed ActorMaterializer from ActorContext

patriknw committed:
Merge pull request #25659 from akka/shardingErrorMessage

Add troubleshooting hint to logging

jrudolph committed:
Fix typo in reference.conf (#25821)

patriknw committed:
fix cluster routing docs (#25810)

* broke when it was moved to separate page

jchapuis committed:
Zip/Unzip support up to 22 ports (#25742)

* Zip/Unzip support up to 22 inputs/outputs respectively

salehkhazaei committed:
Add examples for Fan-in operators #25468

mehdimousavi1995 committed:
example for some sink operators #25468

Examples for takeLast, last and lastOption operators

He-Pin committed:
update headers to regular comment (#25807)

johanandren committed:
Ignore vendor when selecting JVMs  #25835

dwijnand committed:
@DoNotInherit typed.ActorRef (#25851)

ignasi35 committed:
Fix: DNS TTL uses seconds while caching usese millis (#25848)

* Fix: DNS TTL uses seconds while caching uses millis
* Consistently rename ttl vals in companion objects. Add mima filters.
* Rename fields to ttlInZzzz format

dwijnand committed:
Allow OptionVal to hold boxed primitives

Previously OptionVal was restricted to types that have Null as their
bottom type, that is reference types (that extend AnyRef/Object), rather
than also allowing primitives (that extend AnyVal).

By removing this restrictions any type is a valid type argument.

The effect this will have is that primitives will be box, because we
need a way to represent the None case (using the null value).

chbatey committed:
Read port/weight/priority as unsigned shorts for SRVRecords (#25842)

* Read port/weight/priority as unsigned shorts for SRVRecords

* Delete writing code for resource records. Isn't used/tested.

* Mima and review feedback

dwijnand committed:
Dilation cleanups (#25861)

* Avoid calling asInstanceOf in dilated

* Fix the rounding in dilated for Java

* Delegate Java dilated to Scala dilated

* Import scala.concurrent.duration.Duration

dwijnand committed:
Favour Behaviors.receiveMessage in the docs (#25853)

... over Behaviors.receive, but not using the ActorContext binding.

raboof committed:
Introduce 'MemberDowned' member event (#25854)

* Introduce 'MemberDowned' member event

Compatiblity note: MemberEvent is a sealed trait, so it is debatable whether
it is acceptable to introduce a new member.

* Be more conservative (more like leaving), add test

dwijnand committed:
Mini cleanup to stream.ActorMaterializer (#25866)

Do it once, instead of in all 4 call sites.

raboof committed:
Widen timeout waiting for sharding to initialize (#25752) (#25856)

dwijnand committed:
Consistently bind ActorContext to 'context' in the docs (#25852)

* Consistently bind ActorContext to 'ctx' in the docs

... over the lesser uses 'context':

    $ rg 'Behavior[s]?.setup' akka-docs/src/test/scala/docs/actor/TypedActorDocSpec.scala akka-actor-typed-tests/ akka-actor-testkit-typed/ | rg ctx | wc -l
          59
    $ rg 'Behavior[s]?.setup' akka-docs/src/test/scala/docs/actor/TypedActorDocSpec.scala akka-actor-typed-tests/ akka-actor-testkit-typed/ | rg context | wc -l
           8
    $ rg 'Behavior[s]?.receive' akka-docs/src/test/scala/docs/actor/TypedActorDocSpec.scala akka-actor-typed-tests/ akka-actor-testkit-typed/ | rg ctx | wc -l
          66
    $ rg 'Behavior[s]?.receive' akka-docs/src/test/scala/docs/actor/TypedActorDocSpec.scala akka-actor-typed-tests/ akka-actor-testkit-typed/ | rg context | wc -l
           5

* Fix "variable ctx is already defined" warnings

* Replace ctx with context, in docs & doc snippets

    fastmod '\bctx\b' 'context' \
        akka-docs/src/main/paradox/typed
        akka-docs/src/test/scala/docs/actor/TypedActorDocSpec.scala \
        akka-actor-typed-tests/ \
        akka-actor-testkit-typed/ \

* Replace msg with message, in docs & doc snippets

    fastmod '\bmsg\b' 'message' \
        akka-docs/src/test/scala/docs/actor/TypedActorDocSpec.scala \
        akka-actor-typed-tests/ \
        akka-actor-testkit-typed/
        akka-docs/src/main/paradox/typed/

* Fix code alignment

Commit:
Fix typo in lookup of ResolverConfiguration (#25826)

* Fix typo in lookup of ResolverConfiguration

* Set a longer timeout for launching bind

patriknw commented on project/OSGi.scala:
```diff
@@ -91,8 +93,16 @@ object OSGi {
+        scalaParsingCombinatorImport(),
+        sslConfigCoreImport("com.typesafe.sslconfig.ssl.*"),
+        sslConfigCoreImport("com.typesafe.sslconfig.util.*"),
+        "!com.typesafe.sslconfig.akka.*"
```
I guess this change is correct

atooni replied:
I have now rebased and pushed. I can test within the next couple of days if the snapshot works for me in my project.

patriknw reviewed (commented):
@atooni can you rebase this and then we can merge

patriknw committed:
Merge pull request #25868 from dwijnand/unleash-OptionVal

Allow OptionVal to hold boxed primitives

atooni committed:
Merge branch 'osgiSupport' of ssh://github.com/woq-blended/akka into osgiSupport

raboof commented:
@atooni hmm, this diff got quite big, I think because of the last merge of newer akka code into this PR. Perhaps it would be a bit cleaner to rebase instead?

atooni commented:
@raboof I thought I had done a rebase. Over the next couple of days I need to put a container together in our project that contains the modified manifests. I will probably redo the checkout and and create another PR on top of the current master.

atooni commented:
I have now recreated the PR on top of the current master. The new PR is #25906. 
Please reject this PR !!
